src/Units/MissilesModel.java

```diff
@@ -0,0 +1,28 @@
+package Units;
+
+import java.util.Vector;
+
+import Engine.GameAction.ActionType;
+import Terrain.Environment.Terrains;
+import Units.MoveTypes.MoveType;
+import Units.MoveTypes.Tires;
+import Units.MoveTypes.Tread;
+import Units.Weapons.ArtilleryCannon;
+import Units.Weapons.InfantryMGun;
+import Units.Weapons.MissileMissiles;
+import Units.Weapons.RocketRockets;
+import Units.Weapons.WeaponModel;
+
+public class MissilesModel extends UnitModel
```

Comment:
Maybe we should rename this to MobileSAM. "Missiles" is a silly name for a thing that drives, and this would also remove the redundancy in the weapon name.